woocommerce/google-listings-and-ads
# Code refactor: minimum order modals

### Changes proposed in this Pull Request:

In this code refactor PR, we are making some code changes to the minimum order modals (add modal and edit modal):

- Rename "Modal" to "FormModal", to more accurately indicate that the component contains a modal wrapped in a Form with form state in it.
- Rename `value` prop to `initialValues` prop. We shouldn't use `value` because it's not really a controlled component and we can't really dynamically control the value shown in the modal by modify `value`. `initialValues` will be passed directly into the form's `initialValues` props.
- Rename `onChange` prop to `onSubmit` prop. Same reason as above, we shouldn't use `onChange` because it's not really a controlled component and we don't really get the value while the users are changing them in the modal. We only get the value when users submit the form.
- Add JSDoc for the component and its props.

The functionality of the modals remain the same.

The minimum order modals will be used as the reference to modify the add rate modal and edit rate modal later.

### Screenshots:

You can access the add modal and edit modal by clicking on the add button and edit button:

![image](https://user-images.githubusercontent.com/417342/162046032-fc467ddb-b6f8-4b92-a06f-d6f3ea125e4a.png)


### Detailed test instructions:

1. Go to Setup MC or Edit Free Listings, shipping rate section. Make sure you have some shipping rates bigger than 0. Select "yes" for offer free shipping.
2. In the mInimum order card, click on edit button, modify some values, and click on update button. It should work as per existing behavior.
3. In the mInimum order card, click on edit button, and click on delete button. It should work as per existing behavior.
4. Click on the add button, enter some values, and click on add. It should work as per existing behavior.


### Changelog entry
